cmd/kubelet/app/server.go

```diff
@@ -246,6 +247,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&s.MaxPods, "max-pods", 40, "Number of Pods that can run on this Kubelet.")
 	fs.StringVar(&s.DockerExecHandlerName, "docker-exec-handler", s.DockerExecHandlerName, "Handler to use when executing a command in a container. Valid values are 'native' and 'nsenter'. Defaults to 'native'.")
 	fs.StringVar(&s.PodCIDR, "pod-cidr", "", "The CIDR to use for pod IP addresses, only used in standalone mode.  In cluster mode, this is obtained from the master.")
+	fs.StringVar(&s.ResolverConfig, "resolver-config", kubelet.ResolvConfDefault, "Resolver configuration file used as the basis for the container DNS resolution configuration.")
```

Comment:
I think it's more obvious if "" for this flag always means "nothing".  The weird corner case of "oh that means'd Docker's default which implies /etc/resolv.conf".

That said, docker apprently does nto allow blank values, so your very first line item can't work.  I thought there was a way to disable those entirely.  You can pass "." as dns-search to disable search paths, and you can pass 127.0.0.1 to disable nameservers.